tastypie/fields.py

```diff
@@ -763,6 +764,15 @@ def hydrate(self, bundle):
         if value is None:
             return value
 
+        if bundle.obj and isinstance(getattr(bundle.obj.__class__, self.attribute), SingleRelatedObjectDescriptor):
```

Comment:
@SeanHayes oh, thanks! I'll take a look at new pypi version as soon as I can! Great work and many thanks!